### PR TITLE
Adds option to save detailed tensors as host agnostic

### PR DESCRIPTION
**Pull request by GregHattJr**
# Detach and Cast Tensors to CPU for Non-Host Device Readability

### ~~Ticket~~
~~Link to Github Issue~~

### Problem description

This pull request introduces logic that allows tensors to be detached from the computation graph and cast to the CPU before further processing, based on a configurable flag. The main purpose of this change is to improve the ability to read and handle tensors on non-host devices (e.g., reading tensors originally on a GPU from a CPU-based environment). 

#### Reasoning:

- Detaching Tensors: Using .detach() allows tensors to be separated from the computation graph, preventing unnecessary gradient tracking and reducing memory usage. This is useful when tensors need to be utilized for logging, serialization, or analysis without affecting the computational state.
- Casting to CPU: Applying .cpu() transfers tensors to the CPU, which allows them to be read and processed on devices without access to the original device (such as a GPU). This step ensures that data can be read in environments restricted to CPU processing.

#### Impact:

- Allows tensors to be read, processed, and serialized on non-host devices when the config flag is enabled.
- Facilitates workflows where data from GPU operations needs to be accessed in CPU-only contexts for logging, analysis, or other CPU-bound tasks.
- If the config flag is not enabled, the existing functionality remains unchanged, ensuring that current workflows and operations are not impacted.

These changes provide greater flexibility in tensor handling, enabling cross-device use cases while maintaining existing behavior for environments where the configuration is not applied.

### What's changed
- Adds conditional logic within `database.py` to check for the configuration flag and cast to detached torch tensor if flag is enabled.
- Adds logging output if configuration flag is enabled but detailed tensor reporting is not enabled.
- New configuration name (placeholder, still looking for a good name to describe this configuration): `enable_detailed_tensor_report_pytorch_converted`

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/config.hpp [hunk 1]
```diff
@@ -90,6 +91,18 @@ struct Config {
...
         }
 
+        if (
+            name == "enable_detailed_tensor_report_pytorch_converted") {
+            if (not this->attributes.enable_detailed_tensor_report) {
+                if (this->attributes.enable_logging) {
+                    tt::log_warning(
+                        tt::LogAlways,
+                        "Running without detailed tensor report. Please enable detailed tensor report mode if you want "
+                        "to enable saving all tensors as pytorch.");
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/config.hpp:101]: I think we should rephrase. 
"to enable saving all tensors as pytorch" is talking about what the flag does technically, but not what it enables.

Something like:
"Enable detailed tensor report by [DOING X] to enable [WHAT USER WILL BE ABLE TO DO WITH IT]"

Comment (GregHattJr) [reply ttnn/cpp/ttnn/config.hpp]: Changed the wording here. If you have a specific phrasing you'd like let me know and I'll make that change.

**Comment by aidemsined** [comment]
@ayerofieiev-tt @cfjchu @dmakoviichuk-tt   would you please be able to re-review?

File: ttnn/cpp/ttnn/config.hpp [hunk 2]
```diff
@@ -90,6 +91,18 @@ struct Config {
             }
         }
 
+        if (
```
Comment (cfjchu) [inline ttnn/cpp/ttnn/config.hpp:94]: could we clean up this nested condition structure? maybe early return if logging is disabled at the top of the function?

File: ttnn/ttnn/database.py [hunk 3]
```diff
@@ -463,27 +463,52 @@ def load_graph(report_path, operation_id):
...
-        tensor_file_name = tensors_path / f"{tensor.tensor_id}.pt"
-        if tensor_file_name.exists():
-            return
-        torch.save(torch.Tensor(tensor), tensor_file_name)
+        save_tensor_as_torch(tensor, tensor_file_name, DETACH_SAVED_TENSORS)
     else:
         raise ValueError(f"Unsupported tensor type {type(tensor)}")
 
 
+def save_tensor_as_ttnn(tensor, tensor_file_name, detach_saved_tensors):
```
Comment (cfjchu) [inline ttnn/ttnn/database.py:492]: would like to see few unit tests and error handling for these functions please

Comment (GregHattJr) [reply ttnn/ttnn/database.py:492]: Will do.

**Comment by GregHattJr** [comment]
We've decided to take another approach with tensor comparisons so this PR is not longer required.
